src/components/Shared/Markdown/index.tsx

```diff
@@ -0,0 +1,85 @@
+import LexicalAutoLinkPlugin, { URL_MATCHER } from '@components/Shared/Lexical/Plugins/AutoLinkPlugin';
+import type { LensterPublication } from '@generated/lenstertypes';
+import { $createLinkNode, LinkNode } from '@lexical/link';
+import { $convertFromMarkdownString, TEXT_FORMAT_TRANSFORMERS } from '@lexical/markdown';
+import { useLexicalComposerContext } from '@lexical/react/LexicalComposerContext';
+import { ContentEditable } from '@lexical/react/LexicalContentEditable';
+import { HashtagPlugin } from '@lexical/react/LexicalHashtagPlugin';
+import { HistoryPlugin } from '@lexical/react/LexicalHistoryPlugin';
+import { RichTextPlugin } from '@lexical/react/LexicalRichTextPlugin';
+import type { ElementNode, LexicalNode } from 'lexical';
+import { $createTextNode } from 'lexical';
+import type { FC } from 'react';
+import { ERROR_MESSAGE } from 'src/constants';
+
+import MentionsPlugin, { AtSignMentionsRegex } from '../../Shared/Lexical/Plugins/AtMentionsPlugin';
+import { $createMentionNode, MentionNode } from '../Lexical/Nodes/MentionsNode';
+
+const TRANSFORMERS = [
+  ...TEXT_FORMAT_TRANSFORMERS,
+  {
+    dependencies: [MentionNode],
+    export: () => {
+      return null;
+    },
+    regExp: AtSignMentionsRegex,
+    replace: (parentNode: ElementNode, children: Array<LexicalNode>, match: Array<String>) => {
+      // we have to check if the match is a valid mention and then convert it to mentionNode
+      const node = $createMentionNode(match[0].split('@')[1]);
+      parentNode.replace(node);
+      node.select(0, 0);
+      return;
+    },
+    type: 'element'
+  },
+  {
+    dependencies: [LinkNode],
+    export: () => {
+      return null;
+    },
+    importRegExp: /\[([^[]+)]\(([^(]+)\)/,
+    regExp: URL_MATCHER,
+    replace: (textNode: ElementNode, match: Array<string>) => {
+      const [, linkText, linkUrl] = match;
+      const linkNode = $createLinkNode(linkUrl, {
```

Comment:
This is the place for preview transformer where we test for link regex and then convert it to link node.